pytradfri/api/aiocoap_api.py

```diff
@@ -100,8 +101,9 @@ def psk(self) -> str | None:
 
     async def _get_protocol(self) -> Context:
         """Get the protocol for the request."""
-        if self._protocol is None:
-            self._protocol = asyncio.create_task(Context.create_client_context())
+        async with self._protocol_creation_lock:
+            if self._protocol is None:
+                self._protocol = await Context.create_client_context()
         return await self._protocol
```

Comment:
Do we need to await again here?

Reply:
We shouldn't; fixed and also corrected the type annotation. (I hope the `${CUSTOMCLASS} | None` works as intended, I'm not too familiar with the precise type annotations -- would have used `Optional[${CUSTOMCLASS}]` in own code, probabaly)